### PR TITLE
feat(ui): add Tabs component for transcription results display

### PR DESCRIPTION
**Pull request by devodii**

### CODE REVIEW

File: apps/coursition/src/components/transcriptionResults.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        <TextCopier text={srt} title='SRT subtitles' />
+      </TabsContent>
+      <TabsContent value='vtt'>
+        <TextCopier text={vtt} title='VTT subtitles' />
+      </TabsContent>
+    </Tabs>
+  )
+}
+
+const TextCopier = ({ text, title }: { text: string; title: string }) => {
```
Comment (BleedingDev) [inline apps/coursition/src/components/transcriptionResults.tsx:32]: **suggestion:** This should be in separate pull request and the implementation should be probably different. Please remove it from here.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: apps/coursition/src/app/media/page.tsx [hunk 2]
```diff
@@ -116,6 +115,9 @@ export default function Index() {
             )}
           </div>
         )}
+        {status.value === 'done' && (
+          <TranscriptionResults raw={state.raw ?? ''} srt={state.srt ?? ''} vtt={state.vtt ?? ''} />
```
Comment (BleedingDev) [inline apps/coursition/src/app/media/page.tsx:119]: **suggestion:** If some content is optional, it's better to just not pass it at all rather than empty string. And if it's undefined, just don't show the tab at all.

File: apps/coursition/src/components/forgotPassword.tsx [hunk 3]
```diff
@@ -43,7 +46,7 @@ export const ForgotPassword = () => {
       <h2 className='text-[20px] font-semibold'>Reset your password</h2>
       <form className='grid gap-2 w-full max-w-4xl' action={action}>
         <Input label='Email' placeholder='johndoe@gmail.com' id='email' type='email' name='email' />
-        <Button>Request Password reset</Button>
+        <Button disabled={isPending}>Request Password reset</Button>
```
Comment (BleedingDev) [inline apps/coursition/src/components/forgotPassword.tsx:49]: **question:** How is this related to tabs component?

File: libs/ui/design-system/src/lib/tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import * as TabsPrimitive from '@nmit-coursition/ui/primitives/tabs'
+import type { ReactNode } from 'react'
+
+interface TabsProps<Values extends string[]> {
+  triggers: { value: NoInfer<Values[number]>; displayText: string }[]
```
Comment (BleedingDev) [inline libs/ui/design-system/src/lib/tabs.tsx:5]: **question:** Is it possible to have different length, therefore breaking the component? If so, we should alter the API.

**Review by BleedingDev — CHANGES_REQUESTED** [review]

File: libs/ui/design-system/src/lib/tabs.tsx (hunk 4 above)
Comment (devodii) [reply libs/ui/design-system/src/lib/tabs.tsx]: Nicee!!
You overly simplified it.

**Review by BleedingDev — APPROVED** [review]
